### PR TITLE
plan: refine code about maxonerow

### PR DESCRIPTION
**Pull request by hanfei1991**
move maxOneRow out of schema to do further optimize

### CODE REVIEW

File: plan/plan.go [hunk 1]
```diff
@@ -244,14 +252,20 @@ func (p *baseLogicalPlan) buildKeyInfo() {
...
+			p.maxOneRow = p.children[0].(LogicalPlan).MaxOneRow()
+		case *LogicalMaxOneRow, *LogicalExists:
+			p.maxOneRow = true
+	}
+	if len(p.children) == 1 {
 		switch p.self.(type) {
 		case *LogicalExists, *LogicalAggregation, *LogicalProjection:
 			p.basePlan.schema.Keys = nil
 		case *LogicalLock:
 			p.basePlan.schema.Keys = p.basePlan.children[0].Schema().Keys
```
Comment (winoros) [inline plan/plan.go]: don't need basePlan too?

File: plan/logical_plan_test.go [hunk 2]
```diff
@@ -601,6 +601,11 @@ func (s *testPlanSuite) TestSubquery(c *C) {
 			sql:  "select * from t where exists (select s.a from t s having sum(s.a) = t.a )",
 			best: "Join{DataScan(t)->DataScan(s)->Aggr(sum(s.a))->Projection}->Projection",
 		},
+		{
+			// Test MaxOneRow for limit.
+			sql:  "select (select * from (select b from t limit 1) x where x.b = t1.b) from t t1",
+			best: "Join{DataScan(t1)->DataScan(t)->Projection->Limit}(t1.b,x.b)->Projection->Projection",
```
Comment (shenli) [inline plan/logical_plan_test.go]: How can we tell there is a MaxOneRow attribute in the plan?

Comment (hanfei1991) [reply plan/logical_plan_test.go]: Before this test, the plan contains an Apply op.

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@hanfei1991 Please fix CI.

**Pull request merged**